pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/steps/SecretStep.java

```diff
@@ -0,0 +1,152 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package org.jenkinsci.plugins.pipeline.modeldefinition.steps;
+
+import com.google.inject.Inject;
+import com.trilead.ssh2.crypto.Base64;
+import hudson.Extension;
+import hudson.model.Descriptor;
+import jenkins.security.ConfidentialKey;
+import jenkins.security.ConfidentialStore;
+import jenkins.security.CryptoConfidentialKey;
+import net.sf.json.JSONObject;
+import org.apache.commons.lang.ArrayUtils;
+import org.jenkinsci.plugins.workflow.cps.Snippetizer;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution;
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.StaplerRequest;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.crypto.BadPaddingException;
+import javax.crypto.IllegalBlockSizeException;
+import java.io.IOException;
+import java.io.Serializable;
+import java.io.StreamCorruptedException;
+import java.util.Map;
+
+/**
+ * Provides a step to decrypt an encrypted string literal
+ */
+public class SecretStep extends AbstractStepImpl implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private final String text;
+
+    @DataBoundConstructor
+    public SecretStep(String text) {
+        this.text = text;
+    }
+
+    public String getText() {
+        return text;
+    }
+
+    @Extension
+    public static final class DescriptorImpl extends AbstractStepDescriptorImpl {
+
+        private static final int SALT_LENGTH = 8;
+        private static final CryptoConfidentialKey KEY = new CryptoConfidentialKey(SecretStep.class.getName());
+
+        public DescriptorImpl() {
+            super(StepExecutionImpl.class);
+        }
+
+        @Override
+        public String getFunctionName() {
+            return "secret";
+        }
+
+        @Nonnull
+        @Override
+        public String getDisplayName() {
+            return "Provide an encrypted secret that will be decrypted in runtime";
+        }
+
+        @Override
+        public Step newInstance(@CheckForNull StaplerRequest req, @Nonnull JSONObject formData) throws FormException {
+            if (req.findAncestor(Snippetizer.class) != null) {
```

Comment:
Are you sure you do not just want to declare the parameter to be of type `Secret`? Would need a little work in `structs` but probably easy enough.